django/contrib/postgres/indexes.py

```diff
@@ -57,3 +57,37 @@ def get_sql_create_template_values(self, model, schema_editor, using):
 
     def create_sql(self, model, schema_editor):
         return super().create_sql(model, schema_editor, using=' USING gin')
+
+
+class GistIndex(Index):
+    suffix = 'gist'
+    # Allow an index name longer than 30 characters since the suffix is 4
+    # characters (usual limit is 3). Since this index can only be used on
+    # PostgreSQL, the 30 character limit for cross-database compatibility isn't
+    # applicable.
+    max_name_length = 31
+
+    def __init__(self, *, fillfactor=None, buffering=None, **kwargs):
+        self.fillfactor = fillfactor
+        self.buffering = buffering
+        super().__init__(**kwargs)
+
+    def deconstruct(self):
+        path, args, kwargs = super().deconstruct()
+        kwargs['fillfactor'] = self.fillfactor
```

Comment:
I think there's a mistake in `GinIndex.deconstruct()` also... these should be guarded with, e.g. `if self.fillfactor is not None` similar to logic in model fields' `deconstruct()`.

Reply:
I forgot the test when updating, but I just pushed it.